src/components/calculator/Calculator.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+
+import Button from '../common/Button';
+
+import useExpression from '../../hooks/useExpression';
+
+import { ERROR_TEXT } from '../../constants/error';
+
+const Calculator = () => {
+  const [totalText, setTotalText] = useState(0);
+  const [writeExpression, calculateExpression, ResetExpression] = useExpression();
+
+  const handleTextInsert = (event) => {
+    const buttonValue = event.target.value;
+
+    const addText = writeExpression(buttonValue, totalText);
+
+    setTotalText(totalText === 0 || totalText === ERROR_TEXT ? addText : totalText + addText);
+  };
+
+  const handleCalculate = () => {
+    const result = calculateExpression();
+
+    setTotalText(result === Infinity ? ERROR_TEXT : result);
+  };
+
+  const handleAllClear = () => {
+    setTotalText(ResetExpression());
+  };
+
+  return (
+    <div className="calculator">
+      <h1 id="total">{totalText}</h1>
+      <div className="digits flex">
+        <Button className="digit" value="9" onClick={handleTextInsert} />
+        <Button className="digit" value="8" onClick={handleTextInsert} />
+        <Button className="digit" value="7" onClick={handleTextInsert} />
+        <Button className="digit" value="6" onClick={handleTextInsert} />
+        <Button className="digit" value="5" onClick={handleTextInsert} />
+        <Button className="digit" value="4" onClick={handleTextInsert} />
+        <Button className="digit" value="3" onClick={handleTextInsert} />
+        <Button className="digit" value="2" onClick={handleTextInsert} />
+        <Button className="digit" value="1" onClick={handleTextInsert} />
+        <Button className="digit" value="0" onClick={handleTextInsert} />
```

Comment:
저는 이렇게 단순하게 반복해주는 것도 좋다고 생각합니다 😄 
홀수는 키패드 색을 바꾸라던가, 5만 글자크기를 크게 해달라거나 하는 요구사항에 대응이 쉽거든요~!

그래도 그냥 요구사항의 변경이 전혀 없는 미션의 특성상 굳이 리팩토링을 해본다면...
아시겠지만... 요렇게 줄이는 방법도 있을거 같아요! (상수부분은 떼어내는 것도 하면 당연히 좋죠~~ 👍🏻)

 ```suggestion
       { "0123456789".slice("").reverse().map(char => 
          <Button className="digit" key={char} value={char} onClick={handleTextInsert} />) }
```

Reply:
digit와 operation을 상수화 작업하면서 같이 작업해봤습니다!

Reply:
말씀주신대로 상수 작업과 `map`을 이용해서 코드를 줄여봤습니다.
나중에 작업을 진행하려고 했는데 시간이 부족해서 급하게 PR을 올리다가 깜박했네요..🥲